File: EjemploClasePersona/src/ejemploclasepersona/Persona.java

package ejemploclasepersona;

import java.util.Calendar;

public class Persona {
    private String nombre;
    private int anioNacim;
    private char genero;
    
    // Método constructor:
    public Persona(String n,int a,char g) {
        nombre=n;
        anioNacim=a;
        genero=g;
    }
    
    public int calculaEdad() {
        return Calendar.getInstance().get(Calendar.YEAR)-anioNacim;
    }
    
    public String toString() {
        StringBuilder sb=new StringBuilder("");
        sb.append("Nombre: "+nombre+" Año de nacimiento: ");
        sb.append(anioNacim+" Género: "+genero);
        return sb.toString();
    }
}

File: mio/persona.pl
perona(X,B,A).
persona(esteban,1990,asd).

edad(X,N):-persona(X,B,A),N is 2022-B.
toString(X):-persona(X,B,A),write("nombre: "),write(X),
    nl,write("nacimiento: "),write(B),nl
    ,write("GENERO: "),write(A),nl
    ,write("edad: "),edad(X,N),write(N),nl.


